src/material/tooltip/tooltip.ts

```diff
@@ -187,10 +187,7 @@ const MAX_WIDTH = 200;
   },
 })
 export class MatTooltip implements OnDestroy, AfterViewInit {
-  private _overlay = inject(Overlay);
```

Comment:
Do we know what the impact of this is? These are likely already instantiated by the time the tooltip is rendered.

Reply:
The impact is to move a small cost (resolving the injectable) from instantiation time to if/when the tooltip is first shown.

The cost should be small enough that it's not noticeable when showing a tooltip, but does add up during instantiation if lots of MatTooltip instances are being created (imagine a big screen full of content that has tooltips).

Further, most MatTooltip instances will be created but not shown, so on net this reduces the cpu burden of MatTooltip, even if only a smidge.

Reply:
From an example profiler run:

Of the 15.3ms spent instantiating MatTooltip instances, 8.8ms was spent in calls to inject(). This PR removes 1/3 of the inject calls, so we might naively guess that it saves about 3ms, or almost 1/5 of overall MatTooltip instantiation time.

Reply:
This is across many instances, which is sort of the point.

For the page I'm profiling, a big mat-table, the only Directive/component whose factory/constructor took more time is MatCell (3.3% of table render time), and there are a lot more of those (alas chrome's profiler doesn't give exact numbers). MatTooltip's factory took 1.3%, which is quite significant considering how much else is being rendered.

Note: I separately asked Jeremy and Andrew if I could add a lazy version of inject() to the framework and they suggested keep with this pattern instead for the time being.

Reply:
Alright, I think it makes sense for commonly used components like `MatTooltip`, but I don't think that we should apply it everywhere.